samples/vertex-ffp/xgux_extra.h

```diff
@@ -0,0 +1,15 @@
+
+#define MASK(mask, val) (((val) << (ffs(mask)-1)) & (mask))
```

Comment:
This is necessary for ps.inl; the shader compiler should be modified to include this / an XGUX output mode should be added.